src/Throttler.ts

```diff
@@ -63,9 +63,7 @@ export class Throttler {
     private async runInternal(job) {
         this.runningJobPromise = util.sleep(this.delay).then(() => {
             //run the job
-            return Promise.resolve(
-                job()
-            );
+            job();
```

Comment:
this isn't quite right. previously the job could return a promise, and our throttler would wait for that job. Now you're not returning the job results, meaning if it's async, the throttler will continue on as-is. I think you need to `return` the call to job at least. The promise spec mandates that value gets awaited if it's a promise. 

```
return job();
```

Reply:
Oops